cmd/dlt/operatorrole/cmd.go

```diff
@@ -154,43 +118,109 @@ func run(cmd *cobra.Command, argv []string) {
 	}
-	if spin != nil {
-		r.Reporter.Infof("Fetching operator roles for the cluster: %s", clusterKey)
-		spin.Start()
-	}
+	fetchingReporterOutput := "Fetching operator roles for the"
+	if args.prefix == "" {
+		clusterKey = r.GetClusterKey()
+		r.Reporter.Debugf("Loading cluster '%s'", clusterKey)
+		sub, err := r.OCMClient.GetClusterUsingSubscription(clusterKey, r.Creator)
+		if err != nil {
+			if errors.GetType(err) == errors.Conflict {
+				r.Reporter.Errorf("More than one cluster found with the same name '%s'. Please "+
+					"use cluster ID instead", clusterKey)
+				os.Exit(1)
+			}
+			r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
+			os.Exit(1)
+		}
+		if sub != nil {
+			clusterKey = sub.ClusterID()
+		}
+		c, err := r.OCMClient.GetCluster(clusterKey, r.Creator)
+		if err != nil {
+			if errors.GetType(err) != errors.NotFound {
+				r.Reporter.Errorf("Error validating cluster '%s': %v", clusterKey, err)
+				os.Exit(1)
+			} else if sub == nil {
+				r.Reporter.Errorf("Failed to get cluster '%s': %v", r.ClusterKey, err)
+				os.Exit(1)
+			}
+		}
 
-	isHypershift := false
-	if c != nil {
-		isHypershift = c.Hypershift().Enabled()
+		if c != nil && c.ID() != "" {
+			r.Reporter.Errorf("Cluster '%s' is in '%s' state. Operator roles can be deleted only for the "+
+				"uninstalled clusters", c.ID(), c.State())
+			os.Exit(1)
+		}
+		isHypershift := false
+		if c != nil {
+			isHypershift = c.Hypershift().Enabled()
+		} else {
+			subPlanId := sub.Plan().ID()
+			isHypershift = subPlanId == hypershiftSubscriptionPlanId
+		}
+		if spin != nil {
+			fetchingReporterOutput = fmt.Sprintf("%s cluster: %s", fetchingReporterOutput, clusterKey)
+			r.Reporter.Infof("%s", fetchingReporterOutput)
+			spin.Start()
+		}
+		credRequests, err := r.OCMClient.GetCredRequests(isHypershift)
+		if err != nil {
+			r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
+			os.Exit(1)
+		}
+		foundOperatorRoles, _ = r.AWSClient.GetOperatorRolesFromAccountByClusterID(sub.ClusterID(), credRequests)
 	} else {
-		subPlanId := sub.Plan().ID()
-		isHypershift = subPlanId == hypershiftSubscriptionPlanId
-	}
-	credRequests, err := r.OCMClient.GetCredRequests(isHypershift)
-	if err != nil {
-		r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
-		os.Exit(1)
+		if spin != nil {
+			fetchingReporterOutput = fmt.Sprintf("%s prefix: %s", fetchingReporterOutput, args.prefix)
+			r.Reporter.Infof("%s", fetchingReporterOutput)
+			spin.Start()
+		}
+		credRequests, err := r.OCMClient.GetCredRequests(true)
+		if err != nil {
+			r.Reporter.Errorf("Error getting operator credential request from OCM %s", err)
+			os.Exit(1)
+		}
+		foundOperatorRoles, _ = r.AWSClient.GetOperatorRolesFromAccountByPrefix(args.prefix, credRequests)
+		if len(foundOperatorRoles) != 0 {
+			if spin != nil {
+				spin.Stop()
+			}
+			if !confirm.Prompt(true, "You are running delete operation for '%s' prefix."+
```

Comment:
For BYO OIDC, do we have to check that the operator roles are not re-used by another cluster?

Reply:
Would be better, however I couldn't come up with a search query for this just yet. The operator roles prefix attribute is not stored in database, so I'd need to rely on the operator_iam_roles and those are jsonb and the search is not working properly. So after talking to @pvasant we decided to go with a warning for now and work on proper checking later